src/infra/core/src/repos/object_repository_s3.rs

```diff
@@ -192,9 +224,12 @@ where
             .await
             .int_err()?;
 
+        let presigned_request_url = Url::parse(presigned_request.uri()).int_err()?;
+        let url = self.get_external_url(presigned_request_url)?;
```

Comment:
> If we override the host/port, would that potentially invalidate the request signature? Do we have tests that cover presigned URLs with minio?

you're right -- this solution didn't work

> Perhaps its safer to just force smart push/pull to use proxied mode if the storage is not publicly accessible.

Yes, the solution via the "devops" path rather than the "code" path fits better -- checked, it works

Reply:
For the purposes of this PR, I will leave only the refactoring and testing parts

Reply:
PR with a fix for the problem:
- https://github.com/kamu-data/kamu-deploy-example/pull/4